### PR TITLE
[c,z]dotc => [cc,zz]dotc

### PR DESCRIPTION
**Pull request by prj-**
## Pull request purpose

macOS portability

## Detailed changes proposed in this pull request

See also https://github.com/Reference-ScaLAPACK/scalapack/pull/51.

### CODE REVIEW

**Comment by fghoussen** [comment]
CI down, no solution for now (#341). We can't accept more PR. @sylvestre: what do you think? Can you handle this?

**Comment by prj-** [comment]
Just some historical consideration. This portability fix is in FreeFEM for as long as I can remember (https://github.com/FreeFem/FreeFem-sources/blob/07f5919c8ccb6db94cc7fc20fc6b7931c64737e5/3rdparty/arpack/Makefile.am#L72). For two years (https://github.com/FreeFem/FreeFem-sources/blob/6307e79edfb7f7f4f4ee78e3339977066139e5ab/3rdparty/ff-petsc/Makefile#L92), I'm using my fork through SLEPc `--download-arpack` which has this patch, so this has been thoroughly tested already. @joseeroman suggested that I get this merged here so that I don't have to maintain my fork anymore and also this will benefit to (SLEPc + ARPACK) \ FreeFEM users.

**Comment by prj-** [comment]
@fghoussen, I've rebased, could you please run the workflow?

**Comment by sylvestre** [comment]
done

**Comment by fghoussen** [comment]
@prj: can you update the change log?

**Comment by prj-** [comment]
Done.

**Comment by prj-** [comment]
@fghoussen, could you please restart the workflow and approve/comment?

File: CHANGES [hunk 1]
```diff
@@ -1,5 +1,7 @@
 arpack-ng - 3.9.0
  
+* Avoid calling [c|z]dotc for better portability on macOS
```
Comment (fghoussen) [inline CHANGES:3]: You forgot to add your name! (grouping changes by contributors)

Comment (prj-) [reply CHANGES:3]: It didn't seem mandatory, see, e.g., https://github.com/opencollab/arpack-ng/blob/d5f793763f332f2dd0f43063dc38862534ad412d/CHANGES#L241.

Comment (fghoussen) [reply CHANGES:3]: No but it helps to track/look-over changes! As you like

Comment (prj-) [reply CHANGES:3]: I'm OK staying an anonymous John Doe 😄

**Comment by prj-** [comment]
Thank you! I'll get this integrated in SLEPc now and will not have to update my fork anymore.

**Comment by fghoussen** [comment]
Don't hesitate to PR patches back here! :)

**Pull request merged**